README.md
# ELS - Enterprise Linux/EuroLinux Sources - orca
 
## Usage:
  Checkout branch or tag. Use `make sources` to recreate  SOURCE0 archive.


src/ChangeLog
# Generated by Makefile. Do not edit.

2012-10-30 20:26:06 -0400 Joanmarie Diggs <jdiggs@igalia.com>

        One more tweak for bug 687239

M	src/orca/speech.py

2012-10-30 20:13:21 -0400 Joanmarie Diggs <jdiggs@igalia.com>

        Fix for bug 687239 - Traceback seen in Python 3.3 when there are no settings saved

M	src/orca/speech.py

2012-11-12 05:07:46 -0500 Joanmarie Diggs <jdiggs@igalia.com>

        Prep for Orca v3.6.2

M	NEWS
M	README
M	configure.ac

2012-11-11 13:37:36 -0500 Joanmarie Diggs <jdiggs@igalia.com>

        Fix for bug 674526 - Orca sometimes fails to present download "dialog" widgets

M	src/orca/scripts/toolkits/Gecko/script.py

2012-11-11 12:17:24 -0500 Joanmarie Diggs <jdiggs@igalia.com>

        Fix for bug 688057 - A page that hangs orca

M	src/orca/scripts/toolkits/Gecko/script.py

2012-11-11 11:18:06 -0500 Joanmarie Diggs <jdiggs@igalia.com>

        Fix for Orca continuing to speak when speech is disabled via the preferences dialog

M	src/orca/orca_gui_prefs.py

2012-11-07 16:48:13 -0500 Joanmarie Diggs <jdiggs@igalia.com>

        Improve performance of building flat review context

M	src/orca/flat_review.py

2012-11-03 14:13:37 -0400 Frederik Gladhorn <gladhorn@kde.org>

        Fix for bug 687527 - Handle applications returning 0 in minimalIncrement for Value

M	src/orca/script_utilities.py

2012-10-25 11:34:11 +0200 Hendrik Knackstedt <kn.hendrik@gmail.com>

        Updated German translation

M	po/de.po

2012-10-21 16:20:10 +0000 Joanmarie Diggs <jdiggs@igalia.com>

        Use subprocess.Popen() instead of the deprecated os.popen()

M	src/orca/orca_bin.py.in

2012-10-21 16:11:42 +0000 Joanmarie Diggs <jdiggs@igalia.com>

        Use for() instead of map() in cleanup()

M	src/orca/orca_bin.py.in

2012-10-20 19:23:17 +0000 Joanmarie Diggs <jdiggs@igalia.com>

        Fix for bug 686189 - Use importlib.import_module() instead of __import__() and imp.reload() instead of reload()

M	src/orca/app_prefs.py
M	src/orca/orca.py
M	src/orca/script.py
M	src/orca/script_manager.py
M	src/orca/settings.py
M	src/orca/settings_manager.py
M	src/orca/speech.py

2012-10-20 03:06:40 +0900 OKANO Takayoshi <kano@na.rim.or.jp>

        [l10n] Updated Japanese translation

M	po/ja.po

2012-10-20 02:59:34 +0900 Yasumichi Akahoshi <yasumichi@vinelinux.org>

        update ja.po from pot file

M	po/ja.po

2012-10-20 02:49:09 +0900 Yasumichi Akahoshi <yasumichi@vinelinux.org>

        Remove location and wrap from ja.po

M	po/ja.po

2012-10-16 07:35:38 -0400 Joanmarie Diggs <jdiggs@igalia.com>

        Fix to braille email address auto completions in Thunderbird

M	src/orca/scripts/apps/Thunderbird/script.py

2012-10-15 18:33:41 -0400 Joanmarie Diggs <jdiggs@igalia.com>

        Marking as Orca v3.6.2pre

M	README
M	configure.ac

2012-10-15 18:24:58 -0400 Joanmarie Diggs <jdiggs@igalia.com>

        Prep for Orca v3.6.1

M	NEWS
M	README
M	configure.ac

2012-10-15 17:37:13 -0400 Joanmarie Diggs <jdiggs@igalia.com>

        Fix for Bug 675288 - Orca doesn't always announce selection changes in Nautilus 3.6's icon view

M	src/orca/formatting.py
M	src/orca/scripts/apps/nautilus/script.py
M	src/orca/speech_generator.py

2012-10-15 11:55:17 -0400 Joanmarie Diggs <jdiggs@igalia.com>

        Fix for bug 680228 - In Thunderbird 14.0 e-mail address auto completion results are not spoken

M	src/orca/scripts/apps/Thunderbird/script.py

2012-10-13 12:53:00 +0200 Kjartan Maraas <kmaraas@gnome.org>

        Updated Norwegian bokmål translation

M	po/nb.po

2012-09-26 02:53:33 +0200 Frederik Gladhorn <gladhorn@kde.org>

        Use labelOrName for sliders.

M	src/orca/formatting.py

2012-10-03 06:16:01 +0300 Alexander Shopov <ash@kambanaria.org>

        Updated Bulgarian translation

M	po/bg.po

2012-09-26 21:35:01 +0300 Daniel Korostil <ted.korostiled@gmail.com>

        Added uk translation

M	po/uk.po

2012-09-24 22:38:56 -0400 Joanmarie Diggs <jdiggs@igalia.com>

        Markign as Orca v3.6.1pre

M	README
M	configure.ac

2012-09-24 22:37:41 -0400 Joanmarie Diggs <jdiggs@igalia.com>

        Prep for 3.6.0

M	NEWS
M	README
M	configure.ac

2012-09-25 01:28:57 +0300 Rūdolfs Mazurs <rudolfsm@src.gnome.org>

        Updated Latvian translation

M	po/lv.po

2012-09-24 16:46:35 +0200 Petr Kovar <pkovar@redhat.com>

        Update Czech translation by Lucas Lommer

M	po/cs.po

2012-09-22 17:52:40 +0200 Milo Casagrande <milo@ubuntu.com>

        [l10n] Updated Italian translation.

M	po/it.po

2012-09-23 15:05:58 +0200 Simon Bienlein <simon@linux-fuer-blinde.de>

        [l10n] Updated German translation

M	po/de.po

2012-09-22 11:29:01 +0100 Rui Batista <ruiandrebatista@gmail.com>

        Updated Portuguese translation

M	po/pt.po

2012-09-21 20:41:45 +0200 Carles Ferrando <carles.ferrando@gmail.com>

        [l10n]Updated Catalan (Valencian) translation

M	po/ca@valencia.po

2012-09-21 20:41:36 +0200 Gil Forcada <gforcada@gnome.org>

        [l10n] Updated Catalan translation

M	po/ca.po

2012-09-20 13:27:11 +0200 Mario Blättermann <mario.blaettermann@gmail.com>

        [l10n] Updated German translation

M	po/de.po

2012-09-17 14:40:27 -0400 Joanmarie Diggs <jdiggs@igalia.com>

        Marking as v3.6.0pre

M	README
M	configure.ac

2012-09-17 14:35:47 -0400 Joanmarie Diggs <jdiggs@igalia.com>

        Prep for 3.5.92

M	NEWS
M	README
M	configure.ac

2012-09-17 09:51:44 -0300 Rafael Ferreira <rafael.f.f1@gmail.com>

        Updated Brazilian Portuguese translation

M	po/pt_BR.po

2012-09-17 14:11:36 +0200 Daniel Mustieles <daniel.mustieles@gmail.com>

        Updated Spanish translation

M	help/es/es.po

2012-09-16 17:05:10 -0400 Joanmarie Diggs <jdiggs@igalia.com>

        Nudge Gtk+ minimum version down so that it can be built in F18 without jhbuild

M	configure.ac

2012-09-16 13:45:02 -0400 Joanmarie Diggs <jdiggs@igalia.com>

        Bump minimum required version of key dependencies

M	configure.ac

2012-09-16 15:17:40 +0200 Kenneth Nielsen <k.nielsen81@gmail.com>


src/src/orca/platform.py
# Orca
#
# Copyright 2006-2008 Sun Microsystems Inc.
#
# This library is free software; you can redistribute it and/or
# modify it under the terms of the GNU Library General Public
# License as published by the Free Software Foundation; either
# version 2 of the License, or (at your option) any later version.
#
# This library is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
# Library General Public License for more details.
#
# You should have received a copy of the GNU Library General Public
# License along with this library; if not, write to the
# Free Software Foundation, Inc., Franklin Street, Fifth Floor,
# Boston MA  02110-1301 USA.

"""Holds platform-specific settings.
"""

__id__        = "$Id$"
__version__   = "$Revision$"
__date__      = "$Date$"
__copyright__ = "Copyright (c) 2005-2008 Sun Microsystems Inc."
__license__   = "LGPL"

# $ORCA_MAJOR_VERSION.$ORCA_MINOR_VERSION.$ORCA_MICRO_VERSION
#
version     = "2.28.2"

# "--prefix" parameter used when configuring the build.
#
prefix      = "/export/home/wwalker/work/orca/bld"

# The package name (should be "orca").
#
package     = "orca"

# The name of the data directory (usually "share").
#
datadirname = "share"

# The directory where we could find liblouis translation tables.
#
tablesdir = ""


src/src/orca/orca_platform.py
# Orca
#
# Copyright 2006-2008 Sun Microsystems Inc.
#
# This library is free software; you can redistribute it and/or
# modify it under the terms of the GNU Lesser General Public
# License as published by the Free Software Foundation; either
# version 2.1 of the License, or (at your option) any later version.
#
# This library is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
# Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public
# License along with this library; if not, write to the
# Free Software Foundation, Inc., Franklin Street, Fifth Floor,
# Boston MA  02110-1301 USA.

"""Holds platform-specific settings.
"""

__id__        = "$Id$"
__version__   = "$Revision$"
__date__      = "$Date$"
__copyright__ = "Copyright (c) 2005-2008 Sun Microsystems Inc."
__license__   = "LGPL"

# $ORCA_MAJOR_VERSION.$ORCA_MINOR_VERSION.$ORCA_MICRO_VERSION
#
version     = "3.6.3"

# "--prefix" parameter used when configuring the build.
#
prefix      = "/home/jd/orca/bld"

# The package name (should be "orca").
#
package     = "orca"

# The name of the data directory (usually "share").
#
datadirname = "share"

# The directory where we could find liblouis translation tables.
#
tablesdir = "/usr/share/liblouis/tables"


src/src/orca/orca_i18n.py
# Orca
#
# Copyright 2004-2008 Sun Microsystems Inc.
#
# This library is free software; you can redistribute it and/or
# modify it under the terms of the GNU Lesser General Public
# License as published by the Free Software Foundation; either
# version 2.1 of the License, or (at your option) any later version.
#
# This library is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
# Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public
# License along with this library; if not, write to the
# Free Software Foundation, Inc., Franklin Street, Fifth Floor,
# Boston MA  02110-1301 USA.

"""Provides i18n support for orca using the gettext module.  Tells
gettext where to find localized strings and creates an alias, _, that
maps to the gettext.gettext function.  This function will accept a
string and return a localized string for that string.
"""

import os       # to get localdir path
import gettext  # to get gettext (i18n) support

# Alias gettext.gettext to _ and gettext.ngettext to ngettext
#
_ = gettext.gettext
ngettext = gettext.ngettext

# Tell gettext where to find localized strings.
#
localedir = os.path.join ("/home/jd/orca/bld", "share", "locale")
gettext.bindtextdomain ("orca", localedir)
gettext.textdomain("orca")

########################################################################
#                                                                      #
# Utility methods to facilitate easier translation                     #
#                                                                      #
########################################################################

def C_(ctx, s):
    """Provide qualified translatable strings via context."""
    translated = gettext.gettext('%s\x04%s' % (ctx, s))
    if '\x04' in translated:
        # no translation found, return input string
        return s
    return translated
